core/src/main/java/feign/FeignException.java

```diff
@@ -25,27 +26,59 @@
   private static final long serialVersionUID = 0;
   private int status;
   private byte[] content;
+  private Request request;
 
   protected FeignException(int status, String message, Throwable cause) {
     super(message, cause);
     this.status = status;
+    this.request = null;
   }
 
   protected FeignException(int status, String message, Throwable cause, byte[] content) {
     super(message, cause);
     this.status = status;
     this.content = content;
+    this.request = null;
   }
 
   protected FeignException(int status, String message) {
     super(message);
     this.status = status;
+    this.request = null;
   }
 
   protected FeignException(int status, String message, byte[] content) {
     super(message);
     this.status = status;
     this.content = content;
+    this.request = null;
+  }
+
+  protected FeignException(int status, String message, Request request, Throwable cause) {
+    super(message, cause);
+    this.status = status;
+    this.request = checkNotNull(request, "request");
+  }
+
+  protected FeignException(int status, String message, Request request, Throwable cause,
+      byte[] content) {
+    super(message, cause);
+    this.status = status;
+    this.content = content;
+    this.request = checkNotNull(request, "request");
```

Comment:
We should have a real message here.  This will result in an NPE with a message of "request".  Can you change to read something like "request is required"

Reply:
I just grep project to check how checkNotNull invokes and made same. But ok, changed! )